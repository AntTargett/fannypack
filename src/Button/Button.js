import React from 'react';
import PropTypes from 'prop-types';
import * as sharedPropTypes from '../prop-types';

import { Button as StyledButton } from './styled';

const Button = ({ children, color }) => <StyledButton color={color}>{children}</StyledButton>;

Button.propTypes = {
  children: PropTypes.node.isRequired,
  /** Color of the button. Available values: "default", "primary", "secondary", "success", "warning", "danger" */
  color: sharedPropTypes.color
};

Button.defaultProps = {
  color: 'default'
};

Button.propertyControls = {
  children: { type: 'string', title: 'Text' },
  color: { type: 'enum', options: ['default', 'primary', 'success', 'danger'], title: 'Color' }
};

export default Button;
