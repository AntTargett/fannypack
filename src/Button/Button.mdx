---
name: Button
route: /elements/button
menu: Elements
---

import { Playground, PropsTable } from 'docz';
import { Button, Buttons } from '../index';

## Import

`import { Button, Buttons } from 'bumbag'`

## Basic Usage

<Playground>
  <Button>Default button</Button>
</Playground>

## Custom elements

Button, by default, is actually wrapped in a `div` element. See more [here](https://reakit.io/components/button).
However, you are able to change it to a custom element using the `as` prop.

<Playground>
  <Buttons>
    <Button>Default</Button>
    <Button as="a" href="#">Anchor</Button>
  </Buttons>
</Playground>

## Colors

Colors can be added to a button with the `color` prop.

<Playground>
  <Buttons>
    <Button>Default</Button>
    <Button state="primary">Primary</Button>
    <Button state="success">Success</Button>
    <Button state="danger">Danger</Button>
    <Button state="warning">Warning</Button>
  </Buttons>
</Playground>

## Loading

To indicate a button is loading, use the `isLoading` prop.

<Playground>
  <Buttons>
    <Button isLoading>Default</Button>
    <Button state="primary" isLoading>Primary</Button>
    <Button state="primary" isLoading type="link">Primary</Button>
    <Button state="success" isLoading type="outlined">Success</Button>
  </Buttons>
</Playground>

## Outlines

An outline can be placed on a button by assigning `"outlined"` to the `type` prop.

<Playground>
  <Buttons>
    <Button state="primary" type="outlined">Primary</Button>
    <Button state="success" type="outlined">Success</Button>
    <Button state="danger" type="outlined">Danger</Button>
  </Buttons>
</Playground>

## Sizes

Buttons are available in four different sizes and can be added with a `size` prop.

<Playground>
  <Buttons>
    <Button size="small">Small</Button>
    <Button>Default</Button>
    <Button size="medium">Medium</Button>
    <Button size="large">Large</Button>
  </Buttons>
</Playground>

## Links

A button can appear as a link with no styling by adding the `isLink` prop.

<Playground>
  <Button state="primary" type="link">Default</Button>
</Playground>


## Disabled

To indicate a button as disabled, use the `disabled` prop.

<Playground>
  <Buttons>
    <Button disabled>Default</Button>
    <Button state="primary" disabled>Primary</Button>
    <Button state="success" disabled>Success</Button>
    <Button state="danger" disabled>Danger</Button>
    <Button state="warning" disabled>Warning</Button>
  </Buttons>
</Playground>

## Lists

To group a set of buttons, use the `<Buttons>` component.

<Playground>
  <Buttons>
    <Button>Previous</Button>
    <Button>Next</Button>
  </Buttons>
  <Buttons isGrouped marginTop="1rem">
    <Button>Previous</Button>
    <Button>Now</Button>
    <Button>Next</Button>
  </Buttons>
</Playground>

### Submit/Cancel buttons

## `<Button>` Props

<PropsTable of={Button} />

## `<Buttons>` Props

<PropsTable of={Buttons} />
