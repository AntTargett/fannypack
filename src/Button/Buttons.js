// @flow
import React, { type Node } from 'react';

import { Buttons as _Buttons } from './styled';

type Props = {
  children: Node,
  className?: string
};

const Buttons = ({ children, ...props }: Props) => <_Buttons {...props}>{children}</_Buttons>;

Buttons.defaultProps = {
  className: null
};

export default Buttons;
