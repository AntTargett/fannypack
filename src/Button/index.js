export { ThemedButton as Button } from './Button';
export { ThemedButton as default } from './Button';
