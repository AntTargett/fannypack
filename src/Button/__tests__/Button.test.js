import React from 'react';
import render from '../../_utils/tests/render';
import Button from '../Button';
import 'jest-styled-components';

it('renders correctly for a default button', () => {
  const { container } = render(<Button>Test</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a button with a custom element', () => {
  const { container } = render(
    <Button as="a" href="#">
      Test
    </Button>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a button with a palette prop set', () => {
  const { container } = render(<Button palette="primary">Test</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for an outlined button', () => {
  const { container } = render(<Button type="outlined">Test</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a link button', () => {
  const { container } = render(<Button type="link">Test</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a button that is loading', () => {
  const { container } = render(<Button isLoading>Test</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a colored button that is loading', () => {
  const { container } = render(
    <Button isLoading palette="primary">
      Test
    </Button>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a link button that is loading', () => {
  const { container } = render(
    <Button isLoading type="link">
      Test
    </Button>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for an outlined button that is loading', () => {
  const { container } = render(
    <Button isLoading type="outlined">
      Test
    </Button>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a disabled button', () => {
  const { container } = render(<Button disabled>Test</Button>);
  expect(container.firstChild).toMatchSnapshot();
});
