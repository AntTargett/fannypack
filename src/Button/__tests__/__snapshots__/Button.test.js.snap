// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a button that is loading 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #c9c9c9;
  border-radius: 4px;
  color: #333;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2 {
  cursor: not-allowed;
}

.c2:focus {
  outline: unset !important;
}

.c5 {
  width: 1.5rem;
  height: 1.5rem;
}

.c5 path {
  fill: #333;
}

.c4 {
  position: absolute;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 + span {
  opacity: 0;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    style="size: default;"
    type="default"
  >
    <div
      class="c4 c1"
    >
      <svg
        class="c5"
        color="#333"
        version="1.1"
        viewBox="0 0 50 50"
        x="0px"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
        >
          <animatetransform
            attributeName="transform"
            attributeType="xml"
            dur="0.6s"
            from="0 25 25"
            repeatCount="indefinite"
            to="360 25 25"
            type="rotate"
          />
        </path>
      </svg>
    </div>
    <span>
      Test
    </span>
  </button>
</div>
`;

exports[`renders correctly for a button with a custom element 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #c9c9c9;
  border-radius: 4px;
  color: #333;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2:hover {
  background-color: #efefef;
}

.c2:hover:active {
  background-color: #e3e3e3;
}

<div
  class="c0 c1"
>
  <a
    class="c2 c3"
    href="#"
    style="size: default;"
    type="default"
  >
    <span>
      Test
    </span>
  </a>
</div>
`;

exports[`renders correctly for a button with a palette prop set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #112ebb;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #112ebb;
  border: 1px solid #08175d;
  border-radius: 4px;
  color: white;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2:hover {
  background-color: #0f28a4;
}

.c2:hover:active {
  background-color: #0d228c;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    style="size: default;"
    type="default"
  >
    <span>
      Test
    </span>
  </button>
</div>
`;

exports[`renders correctly for a colored button that is loading 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #112ebb;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #112ebb;
  border: 1px solid #08175d;
  border-radius: 4px;
  color: white;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2 {
  cursor: not-allowed;
}

.c2:focus {
  outline: unset !important;
}

.c5 {
  width: 1.5rem;
  height: 1.5rem;
}

.c5 path {
  fill: white;
}

.c4 {
  position: absolute;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 + span {
  opacity: 0;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    style="size: default;"
    type="default"
  >
    <div
      class="c4 c1"
    >
      <svg
        class="c5"
        color="white"
        version="1.1"
        viewBox="0 0 50 50"
        x="0px"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
        >
          <animatetransform
            attributeName="transform"
            attributeType="xml"
            dur="0.6s"
            from="0 25 25"
            repeatCount="indefinite"
            to="360 25 25"
            type="rotate"
          />
        </path>
      </svg>
    </div>
    <span>
      Test
    </span>
  </button>
</div>
`;

exports[`renders correctly for a default button 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #c9c9c9;
  border-radius: 4px;
  color: #333;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2:hover {
  background-color: #efefef;
}

.c2:hover:active {
  background-color: #e3e3e3;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    style="size: default;"
    type="default"
  >
    <span>
      Test
    </span>
  </button>
</div>
`;

exports[`renders correctly for a disabled button 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #c9c9c9;
  border-radius: 4px;
  color: #333;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    disabled=""
    style="size: default;"
    type="default"
  >
    <span>
      Test
    </span>
  </button>
</div>
`;

exports[`renders correctly for a link button 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #c9c9c9;
  border-radius: 4px;
  color: #333;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2 {
  border: 0;
  background: unset;
  box-shadow: unset !important;
  color: #333;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:hover {
  color: #000;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    style="size: default;"
    type="link"
  >
    <span>
      Test
    </span>
  </button>
</div>
`;

exports[`renders correctly for a link button that is loading 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #c9c9c9;
  border-radius: 4px;
  color: #333;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2 {
  border: 0;
  background: unset;
  box-shadow: unset !important;
  color: #333;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:hover {
  color: #000;
}

.c2 {
  cursor: not-allowed;
}

.c2:focus {
  outline: unset !important;
}

.c5 {
  width: 1.5rem;
  height: 1.5rem;
}

.c5 path {
  fill: #fcfcfc;
}

.c4 {
  position: absolute;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 + span {
  opacity: 0;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    style="size: default;"
    type="link"
  >
    <div
      class="c4 c1"
    >
      <svg
        class="c5"
        color="#fcfcfc"
        version="1.1"
        viewBox="0 0 50 50"
        x="0px"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
        >
          <animatetransform
            attributeName="transform"
            attributeType="xml"
            dur="0.6s"
            from="0 25 25"
            repeatCount="indefinite"
            to="360 25 25"
            type="rotate"
          />
        </path>
      </svg>
    </div>
    <span>
      Test
    </span>
  </button>
</div>
`;

exports[`renders correctly for an outlined button 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #c9c9c9;
  border-radius: 4px;
  color: #333;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2 {
  background-color: unset;
  border: 1px solid #fcfcfc;
  color: #fcfcfc;
}

.c2:hover {
  color: #333;
}

.c2:hover {
  background-color: #efefef;
}

.c2:hover:active {
  background-color: #e3e3e3;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    style="size: default;"
    type="outlined"
  >
    <span>
      Test
    </span>
  </button>
</div>
`;

exports[`renders correctly for an outlined button that is loading 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  margin: 0;
  padding: 0;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #c9c9c9;
  border-radius: 4px;
  color: #333;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 500;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2 {
  background-color: unset;
  border: 1px solid #fcfcfc;
  color: #fcfcfc;
}

.c2:hover {
  color: #333;
}

.c2 {
  cursor: not-allowed;
}

.c2:focus {
  outline: unset !important;
}

.c5 {
  width: 1.5rem;
  height: 1.5rem;
}

.c5 path {
  fill: #fcfcfc;
}

.c4 {
  position: absolute;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 + span {
  opacity: 0;
}

<div
  class="c0 c1"
>
  <button
    class="c2 c3"
    style="size: default;"
    type="outlined"
  >
    <div
      class="c4 c1"
    >
      <svg
        class="c5"
        color="#fcfcfc"
        version="1.1"
        viewBox="0 0 50 50"
        x="0px"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
        >
          <animatetransform
            attributeName="transform"
            attributeType="xml"
            dur="0.6s"
            from="0 25 25"
            repeatCount="indefinite"
            to="360 25 25"
            type="rotate"
          />
        </path>
      </svg>
    </div>
    <span>
      Test
    </span>
  </button>
</div>
`;
